### DISCUSSION TITLE
Can Pyright update references when renaming a file?

### DISCUSSION THREAD

**Discussion by fredrikaverpil** [microsoft/pyright#5253] [Q&A] [answered]
I'm on Neovim (pyright) and I miss the ability to rename/move files and have the LSP update all references in other files which works fine in vscode (pylance).

Is it possible to somehow use Pyright for this?

**Comment by erictraut** [comment, marked as answer]
Pyright doesn't support "rename/remove files". This is an advanced LSP feature that is supported by Pylance, Microsoft's premier language server for VS Code.

While pyright has some basic language server features, it's focused on core type checking, and we have no plans to add new language server features to pyright. If you want more advanced LSP features, you have the option of switching to VS Code and using Pylance or building language server functionality yourself on top of the core type checking capabilities of pyright.
